src/util/domain-formatter.ts

```diff
@@ -130,13 +130,43 @@ export function removeTLD(hostname: string): string {
   return hostname.slice(0, lastDot);
 }
 
+// Helper function to check if an address is an IPv4 bridged to IPv6
+function isIPv4BridgedToIPv6(address: string): boolean {
+  return /^::ffff:([0-9a-f]{8})$/i.test(address);
+}
+
+// Add a new function to handle IPv4 bridged to IPv6 addresses
+export function formatIPv4BridgedToIPv6(address: string): string {
+  if (!isIPv4BridgedToIPv6(address)) {
+    assert(net.isIPv4(address), "Illegal argument. Must be an IPv4 address!");
+  }
+
+
+  assert(net.isIPv4(address), "Illegal argument. Must be an IPv4 address!");
+
+  // Convert IPv4 address to its hexadecimal representation
+  const hexParts = address.split(".").map(part => parseInt(part).toString(16).padStart(2, "0"));
+  const ipv6Part = `::ffff:${hexParts.join("")}`;
+
+  // Convert the hexadecimal representation to the standard IPv6 format
+  return ipv6Part.replace(/(.{4})(.{4})$/, "$1:$2");
+}
+
 export function enlargeIPv6(address: string): string {
   assert(net.isIPv6(address), "Illegal argument. Must be ipv6 address!");
 
-  // we are not supporting ipv4-mapped ipv6 addresses here
-  assert(!address.includes("."), "ipv4-mapped ipv6 addresses are currently unsupported!");
+  // Handling IPv4-mapped IPv6 addresses
+  if (address.includes(".")) {
+    return address; // Return as is because it's an IPv4-mapped address
+  }
+
+  const ipv4MappedIPv6Regex = /^::ffff:(\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3})$/;
```

Comment:
Might want to add an `i` after the slash to make it case-insensitive.

Reply:
made this change, if you could please review?